Source/Core/Common/MemoryUtil.cpp

```diff
@@ -38,9 +39,11 @@ void* AllocateExecutableMemory(size_t size)
 #if defined(_WIN32)
   void* ptr = VirtualAlloc(nullptr, size, MEM_COMMIT, PAGE_EXECUTE_READWRITE);
 #else
-  void* ptr =
-      mmap(nullptr, size, PROT_READ | PROT_WRITE | PROT_EXEC, MAP_ANON | MAP_PRIVATE, -1, 0);
-
+  int map_flags = MAP_ANON | MAP_PRIVATE;
+#if defined(_M_ARM_64) && defined(__APPLE__)
+  map_flags |= MAP_JIT;
```

Comment:
Just for future reference: if ``MAP_JIT`` is set on x86_64 too in the future, it is necessary to do a version check as older macOS will crash if the flag is set.

An example can be found [here](https://github.com/dolphin-emu/dolphin/pull/8909#issuecomment-766305089). (I think you might be able to use ``__builtin_available(macOS 10.14, *)`` instead of fetching the version from ``utsname``?)

With many thanks to @ryanmcgrath for the heads up.

Reply:
> I think you might be able to use __builtin_available(macOS 10.14, *) instead of fetching the version from utsname?
That would be a compile time check for availablity of MAP_JIT, you'd want the runtime check.

Reply:
[__builtin_available is a runtime check](https://developer.apple.com/videos/play/wwdc2017/411/?time=352).

Reply:
Ill add this check just so it doesn't get forgotten if someone tries to add x86_64 JIT page support.